modules/tensor_mechanics/src/actions/TensorMechanicsAction.C

```diff
@@ -809,6 +766,12 @@ TensorMechanicsAction::actGatherActionParameters()
   if (_current_task == "validate_coordinate_systems" && getParam<bool>("add_variables"))
   {
     auto actions = _awh.getActions<TensorMechanicsAction>();
+
+    // is the current action teh first one in the warehouse?
+    if (actions.empty())
+      mooseError("Internal error");
+    _is_primary_action = (actions[0] == this);
```

Comment:
Is there anything the user can do via the input file to handle this situation? If yes, I'd suggest a more verbose error message. Also spelling correction need for the comment

Is line 773 enforcing a 'first come first served' sort of policy with the setting of `_is_primary_action`? Is that prudent? In any case, would you explain for me how this line work?

Reply:
This should never happen. It basically checks if there is any TM action at all... but of course there is at least one (the one that is performing this check). Nothing the user can do about this. This is purely intended to fail CI testing in case a dev does something super weird.